src/app/shared/file-input/file-input.component.ts

```diff
@@ -0,0 +1,71 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { AlertController } from '@ionic/angular';
+import { TranslateService } from '@ngx-translate/core';
+import { v4 as uuidv4 } from 'uuid';
+
+@Component({
+  selector: 'fb-file-input',
+  templateUrl: 'file-input.component.html',
+  styleUrls: ['file-input.component.scss']
+})
+export class FileInputComponent implements OnInit {
+
+  fileName: string;
+
+  @Input()
+  uploadSuccessful = false;
+
+  progress = 0;
+
+  @Output()
+  fileContent = new EventEmitter<string | ArrayBuffer>();
+
+  @Output()
+  onFileSelectEvent = new EventEmitter<File>();
+
+  @Output()
+  deleteEvent = new EventEmitter<string>();
+
+  constructor(
+    private alertController: AlertController,
+    private translate: TranslateService) {
+  }
+
+  ngOnInit() {
+  }
+
+  async onFileSelect(event: any) {
+    const fileType = (event.target as HTMLInputElement).files[0].name.slice(-3);
+    if (fileType != "igc") {
```

Comment:
👍 👍

Reply:
Maybe we should lowercase the fileType in the check? 
I just noticed that my sample .igc file I downloaded from xContest (https://www.xcontest.org/switzerland/en/flights/detail:robin.eyer/23.04.2021/16:08) has an uppercase .IGC ending, in this case the upload doesn't work without renaming. Not sure how often this will really happen, but doesn't hurt to lowercase it just in case.

Reply:
👍 Ohhh very good input.
Done